### PR TITLE
Add docstrings to data and drop verbose helper

### PR DESCRIPTION
**Pull request by bhancock8**
- Add docstrings to the classes in data.py
- Drop the split_data() helper. It's nice, but its functionality is _mostly_ redundant with scikit-learn's train_test_split() and it's a lot of lines of code for something that's not core to the library's value.

**Test plan:**
- pydocstyle passes
- unit tests pass

### CODE REVIEW

File: snorkel/classification/data.py [hunk 1]
```diff
@@ -27,17 +12,24 @@
...
+    name
+        The name of the dataset (e.g., this will be used to report metrics on a
+        per-dataset basis)
+    split
+        The name of the split that the data in this object represents
+    X_dict
+        A map from field name to values (e.g., {"tokens": ..., "uids": ...})
+    Y_dict
+        A map from task name to its corresponding set of labels
+
```
Comment (henryre) [inline snorkel/classification/data.py]: Add Attributes (just "see above" is fine)

File: snorkel/classification/data.py [hunk 2]
```diff
@@ -87,18 +90,16 @@ def collate_dicts(batch: List[Batch]) -> Batch:
...
-    :param split: str, optional
-    :param collate_fn: the function that merges a list of samples to form a
-    mini-batch, defaults to collate_dicts
-    :param collate_fn: function, optional
+    """A DataLoader that uses the appropriate collate_fn for a `DictDataset`.
+
+    Parameters
+    ----------
+    dataset
+        A DictDataset to wrap
```
Comment (henryre) [inline snorkel/classification/data.py]: Avoid having param descriptions that just repeat the type

File: snorkel/classification/data.py [hunk 3]
```diff
@@ -87,18 +90,16 @@ def collate_dicts(batch: List[Batch]) -> Batch:
...
+    collate_fn
+        The collate function to use when combining multiple indexed examples for a
+        single batch. Usually the default collate_dicts() method should be used, but
+        it can be overriden if you want to use different collate logic.
```
Comment (henryre) [inline snorkel/classification/data.py]: kwargs?

**Comment by bhancock8** [comment]
Re: split_data(): It was mostly a utility for general data prep, not core code. I think it's used in the tutorials, but we can update them.

**Pull request merged**